From 85c31977d336e3e7e8cdea607c9de07ffa557ce1 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <xiaowen.liu@freescale.com>
Date: Mon, 13 Apr 2015 16:48:37 +0800
Subject: [PATCH 34/45] MA-6484 fix Setting: User can't add new user or
 restricted profile.

The root cause is 32bit casted to 64bit issue.
The casting should happen before calculation.

Signed-off-by: Liu Xiaowen <xiaowen.liu@freescale.com>
---
 services/core/java/com/android/server/pm/UserManagerService.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/pm/UserManagerService.java b/services/core/java/com/android/server/pm/UserManagerService.java
index 7dd4745..cf21b0f 100644
--- a/services/core/java/com/android/server/pm/UserManagerService.java
+++ b/services/core/java/com/android/server/pm/UserManagerService.java
@@ -655,7 +655,7 @@ public class UserManagerService extends IUserManager.Stub {
     private static final boolean isNoFreeSpace() {
         final String path = Environment.getDataDirectory().getPath();
         StatFs fileStats = new StatFs(path);
-        long restM = fileStats.getAvailableBlocks() * fileStats.getBlockSize();
+        long restM = (long)fileStats.getAvailableBlocks() * (long)fileStats.getBlockSize();
         Slog.w(LOG_TAG, "remain size for data is " + restM);
         if ( restM < MIN_AVAILABLE_SPACE)
             return true;
-- 
1.8.0

